discv5/discv5-wire.md

```diff
@@ -49,187 +51,107 @@ request/response and 1s for the handshake.
 
-`id_sign(data)` creates a signature over `data` using the node's static private key. The
-signature is encoded as the 64-byte array `r || s`, i.e. as the concatenation of the
-signature values.
+Decrypting the masked header data works as follows: The recipient constructs an AES/CTR
+stream cipher using its own node ID (`dest-id`) as the key and taking the IV from the
+packet. It can then decrypt the `static-header` and verify that `protocol-id` matches. If
+it does, the recipient can read `authdata-size` and finally decrypt the remaining
+`authdata`.
 
-`ecdh(pubkey, privkey)` creates a secret through elliptic-curve Diffie-Hellman key
-agreement. The public key is multiplied by the private key to create a secret ephemeral
-key `eph = pubkey * privkey`. The 33-byte secret output is `y || eph.x` where `y` is
-`0x02` when `eph.y` is even or `0x03` when `eph.y` is odd.
+Implementations should not respond to a packet with mismatching `protocol-id`.
 
-## Packet Encoding
+The `flag` field of the header identifies the kind of packet and determines the encoding
+of `authdata`, which differs depending on the packet type.
 
-All regular packets except WHOAREYOU start with a fixed-size `tag`. For a packet sent by
-node A to node B:
+### Ordinary Message Packet (`flag = 0`)
 
-    tag              = xor(sha256(dest-node-id), src-node-id)
-    dest-node-id     = 32-byte node ID of B
-    src-node-id      = 32-byte node ID of A
+For message packets, the `authdata` section is just the 96-bit AES/GCM nonce:
 
-The recipient can recover the sender's ID by performing the same calculation in reverse.
+    authdata      = nonce
+    authdata-size = 12
 
-    src-node-id      = xor(sha256(dest-node-id), tag)
+![message packet layout](./img/message-packet-layout.png)
 
-The encoding of the 'random packet', sent if no session keys are available, is:
+### WHOAREYOU Packet (`flag = 1`)
 
-    random-packet    = tag || rlp_bytes(auth-tag) || random-data
-    auth-tag         = 12 random bytes unique to message
-    random-data      = at least 44 bytes of random data
+In WHOAREYOU packets, the `authdata` section contains information for the verification
+procedure. The `message` field of WHOAREYOU packets is always empty.
 
-The WHOAREYOU packet, used during the handshake, is encoded as follows:
+    authdata      = request-nonce || id-nonce || enr-seq
+    authdata-size = 52
+    request-nonce = uint96    -- nonce of request packet that couldn't be decrypted
+    id-nonce      = uint256   -- random bytes
+    enr-seq       = uint64    -- ENR sequence number of the requesting node
 
-    whoareyou-packet = magic || [token, id-nonce, enr-seq]
-    magic            = sha256(dest-node-id || "WHOAREYOU")
-    token            = auth-tag of request
-    id-nonce         = 32 random bytes
-    enr-seq          = highest ENR sequence number of node A known on node B's side
+![whoareyou packet layout](./img/whoareyou-packet-layout.png)
```

Comment:
On the diagram there is a capture `AESGCM nonce or message packet` 
What the `... or message packet` stands for?

Reply:
It says 'AESGCM nonce **of** message packet', but maybe it's hard to read because of the color.

Reply:
Oh right, sorry!